common/src/java/org/apache/hadoop/hive/conf/HiveConfUtil.java

```diff
@@ -199,16 +199,16 @@ public static void updateJobCredentialProviders(Configuration jobConf) {
     if (credstorePassword != null) {
       String execEngine = jobConf.get(ConfVars.HIVE_EXECUTION_ENGINE.varname);
 
-      if ("mr".equalsIgnoreCase(execEngine)) {
-        // if the execution engine is MR set the map/reduce env with the credential store password
+      if ("mr".equalsIgnoreCase(execEngine) || "tez".equalsIgnoreCase(execEngine)) {
+        // if the execution engine is MR/Tez set the map/reduce env with the credential store password
 
         Collection<String> redactedProperties =
             jobConf.getStringCollection(MRJobConfig.MR_JOB_REDACTED_PROPERTIES);
-
         Stream.of(
             JobConf.MAPRED_MAP_TASK_ENV,
             JobConf.MAPRED_REDUCE_TASK_ENV,
-            MRJobConfig.MR_AM_ADMIN_USER_ENV)
+            MRJobConfig.MR_AM_ADMIN_USER_ENV,
+            "tez.am.launch.env")
```

Comment:
Is this credential required for tez.task.launch.env ?

Reply:
thanks @nareshpr, I'll double-check, I tested this patch on a cluster as a POC, and I saw a simple map/reduce dag worked
also, I'm going to decide how to use this as referring to TezConfiguration.TEZ_AM_LAUNCH_ENV, which is not visible currently in hive-common